### PR TITLE
libvirt.py: Add ignore_str param to check_logfile

### PR DESCRIPTION
**Pull request by liuyd96**
Add ignore_str param to check_logfile. Ignore specific str if needed.

Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -3830,10 +3830,13 @@ def check_logfile(search_str, log_file, str_in_log=True,
     :param runner_on_target:  Remote runner
     :param ignore_status: True to return False on failure,
                           False to raise exception on failure
+    :param ignore_str: Ignore str
     :raise: test.fail when the result is not expected
     :return: True if check is successful, otherwise False
     """
     cmd = "grep -E '%s' %s" % (search_str, log_file)
+    if len(ignore_str) != 0:
```
Comment (chunfuwen) [inline virttest/utils_test/libvirt.py:3838]: Do we need add ignore_statu=True as precondition?

Comment (liuyd96) [reply virttest/utils_test/libvirt.py:3838]: Hi, @chunfuwen 

> Do we need add ignore_statu=True as precondition?

No. I think the ignore_str and ignore_status should be irrelevant.
No matter ignore_status is True or False, we should also supports ignore_str to filter out unexpected strs.

**Review by chunfuwen — APPROVED** [review]
LGTM

**Comment by liuyd96** [comment]
@chunfuwen @Yingshun Gently ping ~

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -3830,10 +3830,13 @@ def check_logfile(search_str, log_file, str_in_log=True,
...
+        cmd += f" -v {ignore_str}"
```
Comment (Yingshun) [inline virttest/utils_test/libvirt.py:3839]: Adding '-v ignore_str' after 'log_file' will cause a failure. Are you going to add `cmd += f"| grep -v {ignore_str}"`?

**Comment by Yingshun** [comment]
BTW, would you please upload test results?

**Comment by liuyd96** [comment]
This fix should work with https://github.com/autotest/tp-libvirt/pull/4472
Before this fix
```
 (1/1) type_specific.io-github-autotest-libvirt.libvirtd.libvirtd.check_journal: FAIL: The string 'error :' is included in /var/log/libvirt/libvirtd.log (44.99 s)
```
Now
```
 (1/1) type_specific.io-github-autotest-libvirt.libvirtd.libvirtd.check_journal: PASS (46.48 s)
```

**Review by Yingshun — APPROVED** [review]

**Pull request merged**